### DISCUSSION TITLE
Custom Scripts - Objects Missing Attributes

### DISCUSSION THREAD

**Discussion by adparis99** [netbox-community/netbox#10668] [Q&A] [answered]
Hello, as the title suggests I'm trying to perfect a custom script that pulls the attributes of different objects, namely IP Addresses, Devices, and Interfaces. However, I notice that when I use Interface.objects.filter() to create a list of interfaces that they are missing some of the attributes I'm looking for, such as the untagged or tagged VLANS. I assume there is a way to get this info as doing the API call through pynetbox or on the web GUI provides the results I'm accustomed to. Just curious if there's a way to access this info in custom scripts that I am missing.

**Comment by candlerb** [comment, marked as answer]
Custom scripts use the Django ORM directly, which is different to the REST API (which pynetbox uses).

Use the [Netbox shell](https://docs.netbox.dev/en/stable/administration/netbox-shell/) to explore the attributes available on an object.  Search for "[nbshell](https://github.com/netbox-community/netbox/discussions?discussions_q=nbshell)" in the discussion archives for some examples.
